From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@debian.org>
Date: Mon, 28 Nov 2016 14:47:05 +0100
Subject: Use RuntimeDirectory instead of tmpfiles.d

This reverts part of commit 32123f6bb2ebc4f9ac7865945cc85a9c9b903dc6.

A simple directive is a much lighter solution to the same problem, and
automatically follows the specified User.
---
 conf/Makefile.am                    |  2 +-
 conf/tmpfiles.d/Makefile.am         | 40 -------------------------------------
 conf/tmpfiles.d/corosync-qnetd.conf |  1 -
 configure.ac                        | 10 +---------
 corosync.spec.in                    |  6 +-----
 init/corosync-qdevice.service.in    |  3 +++
 init/corosync-qnetd.service.in      |  1 +
 7 files changed, 7 insertions(+), 56 deletions(-)
 delete mode 100644 conf/tmpfiles.d/Makefile.am
 delete mode 100644 conf/tmpfiles.d/corosync-qnetd.conf

diff --git a/conf/Makefile.am b/conf/Makefile.am
index 6382386..807b6cc 100644
--- a/conf/Makefile.am
+++ b/conf/Makefile.am
@@ -70,4 +70,4 @@ dbusdir			= $(sysconfdir)/dbus-1/system.d
 dbus_DATA		= corosync-signals.conf
 endif
 
-SUBDIRS = logrotate tmpfiles.d
+SUBDIRS = logrotate
diff --git a/conf/tmpfiles.d/Makefile.am b/conf/tmpfiles.d/Makefile.am
deleted file mode 100644
index 653129a..0000000
--- a/conf/tmpfiles.d/Makefile.am
+++ /dev/null
@@ -1,40 +0,0 @@
-# Copyright (c) 2016 Red Hat, Inc.
-#
-# Authors:Jan Friesse (jfriesse@redhat.com)
-#
-# This software licensed under BSD license, the text of which follows:
-#
-# Redistribution and use in source and binary forms, with or without
-# modification, are permitted provided that the following conditions are met:
-#
-# - Redistributions of source code must retain the above copyright notice,
-#   this list of conditions and the following disclaimer.
-# - Redistributions in binary form must reproduce the above copyright notice,
-#   this list of conditions and the following disclaimer in the documentation
-#   and/or other materials provided with the distribution.
-# - Neither the name of the MontaVista Software, Inc. nor the names of its
-#   contributors may be used to endorse or promote products derived from this
-#   software without specific prior written permission.
-#
-# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
-# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
-# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
-# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
-# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
-# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
-# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
-# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
-# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
-# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
-# THE POSSIBILITY OF SUCH DAMAGE.
-
-MAINTAINERCLEANFILES    = Makefile.in
-
-EXTRA_DIST		= corosync-qnetd.conf
-
-if BUILD_QNETD
-if INSTALL_SYSTEMD
-tmpfilesdirdir		= ${TMPFILESDIR}
-tmpfilesdir_DATA	= corosync-qnetd.conf
-endif
-endif
diff --git a/conf/tmpfiles.d/corosync-qnetd.conf b/conf/tmpfiles.d/corosync-qnetd.conf
deleted file mode 100644
index f169d36..0000000
--- a/conf/tmpfiles.d/corosync-qnetd.conf
+++ /dev/null
@@ -1 +0,0 @@
-d /run/corosync-qnetd 0755 coroqnetd coroqnetd
diff --git a/configure.ac b/configure.ac
index e9c7d09..b6ed21b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -212,8 +212,7 @@ AC_CONFIG_FILES([Makefile
 		 Doxyfile
 		 conf/corosync.conf.example
 		 conf/corosync.conf.example.udpu
-		 conf/logrotate/Makefile
-		 conf/tmpfiles.d/Makefile])
+		 conf/logrotate/Makefile])
 
 for daemon in corosync corosync-notifyd corosync-qdevice corosync-qnetd; do
     AC_CONFIG_FILES([init/$daemon init/$daemon.service])
@@ -413,11 +412,6 @@ AC_ARG_WITH([logrotatedir],
 	[ LOGROTATEDIR="$withval" ],
 	[ LOGROTATEDIR="$sysconfdir/logrotate.d" ])
 
-AC_ARG_WITH([tmpfilesdir],
-	[  --with-tmpfilesdir=DIR   : path to tmpfiles.d configuration files directory. ],
-	[ TMPFILESDIR="$withval" ],
-	[ TMPFILESDIR="/lib/tmpfiles.d" ])
-
 AC_ARG_ENABLE([snmp],
 	[  --enable-snmp                   : SNMP protocol support ],
 	[ default="no" ])
@@ -756,7 +750,6 @@ INITWRAPPERSDIR=$(expand $INITWRAPPERSDIR)
 AC_SUBST([INITWRAPPERSDIR])
 AC_SUBST([LOGDIR])
 AC_SUBST([LOGROTATEDIR])
-AC_SUBST([TMPFILESDIR])
 
 AC_SUBST([SOMAJOR])
 AC_SUBST([SOMINOR])
@@ -801,7 +794,6 @@ AC_MSG_RESULT([  System init.d directory  = ${INITDDIR}])
 AC_MSG_RESULT([  System systemd directory = ${SYSTEMDDIR}])
 AC_MSG_RESULT([  System upstart directory = ${UPSTARTDIR}])
 AC_MSG_RESULT([  System init wraps dir    = ${INITWRAPPERSDIR}])
-AC_MSG_RESULT([  System tmpfiles.d        = ${TMPFILESDIR}])
 AC_MSG_RESULT([  Log directory            = ${LOGDIR}])
 AC_MSG_RESULT([  Log rotate directory     = ${LOGROTATEDIR}])
 AC_MSG_RESULT([  corosync config dir      = ${COROSYSCONFDIR}])
diff --git a/corosync.spec.in b/corosync.spec.in
index 84fb507..cfe0dc4 100644
--- a/corosync.spec.in
+++ b/corosync.spec.in
@@ -126,8 +126,7 @@ export rdmacm_LIBS=-lrdmacm \
 %endif
 	--with-initddir=%{_initrddir} \
 	--with-systemddir=%{_unitdir} \
-	--with-upstartdir=%{_sysconfdir}/init \
-	--with-tmpfilesdir=%{_tmpfilesdir}
+	--with-upstartdir=%{_sysconfdir}/init
 
 make %{_smp_mflags}
 
@@ -168,8 +167,6 @@ install -m 644 init/corosync-qnetd.sysconfig.example \
 %if %{with systemd}
 sed -i -e 's/^#User=/User=/' \
    %{buildroot}%{_unitdir}/corosync-qnetd.service
-sed -i -e 's/root/coroqnetd/g' \
-   %{buildroot}%{_tmpfilesdir}/corosync-qnetd.conf
 %else
 sed -i -e 's/^COROSYNC_QNETD_RUNAS=""$/COROSYNC_QNETD_RUNAS="coroqnetd"/' \
    %{buildroot}%{_sysconfdir}/sysconfig/corosync-qnetd
@@ -495,7 +492,6 @@ fi
 %{_unitdir}/corosync-qnetd.service
 %dir %{_datadir}/corosync
 %{_datadir}/corosync/corosync-qnetd
-%{_tmpfilesdir}/corosync-qnetd.conf
 %else
 %{_initrddir}/corosync-qnetd
 %endif
diff --git a/init/corosync-qdevice.service.in b/init/corosync-qdevice.service.in
index 001cb75..33104a3 100644
--- a/init/corosync-qdevice.service.in
+++ b/init/corosync-qdevice.service.in
@@ -10,6 +10,9 @@ EnvironmentFile=-@CONFIGDIR@/corosync-qdevice
 ExecStart=@sbindir_expanded@/corosync-qdevice -f $COROSYNC_QDEVICE_OPTIONS
 Type=simple
 Restart=on-abnormal
+RuntimeDirectory=corosync-qdevice
+# Don't rely on the socket persmissions (man 7 unix)
+RuntimeDirectoryMode=0700
 
 [Install]
 WantedBy=multi-user.target
diff --git a/init/corosync-qnetd.service.in b/init/corosync-qnetd.service.in
index 1a04960..7e550d2 100644
--- a/init/corosync-qnetd.service.in
+++ b/init/corosync-qnetd.service.in
@@ -11,6 +11,7 @@ ExecStart=@bindir_expanded@/corosync-qnetd -f $COROSYNC_QNETD_OPTIONS
 Type=simple
 Restart=on-abnormal
 User=coroqnetd
+RuntimeDirectory=corosync-qnetd
 
 [Install]
 WantedBy=multi-user.target
